modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -1800,6 +1800,16 @@ public T getCell(int index) {
         return accumCell;
     }
 
+    /**
+     * The amount of pixels by which to adjust the position when the the scroll bar track is clicked.
+     *
+     * @return the value in pixels
+     */
+    public double getBlockIncrement() {
+        // For reasons of backward compatibility, we use the cell length of the empty cell (cell index -1)
+        return getCellLength(-1);
```

Comment:
How would that work when cell heights are different for each cell?

What if the cell height is much larger than the viewport height (as in, 10 times larger?)

Reply:
I don't like that either, but as I said I didn't want to change it because of backwards compatibility. 

----
I would like to make it configurable or even make the complete scrolling behavior interchangeable.
Because, the actual scrolling behavior is still not consistent. 

If multiple cells are visible, the cells are currently aligned (when clicking on the scrollbar) so that they are either exactly at the top or exactly at the bottom. If only one cell is visible, the scrolling behavior changes to the applying of pixel delta.

For example, if you have two large cells and a small ViewPort so that only one cell is visible.
1) If you click on the thumb below the scroll bar, the ListView is scrolled by a pixel delta so that two cells can become visible. 
2) If you click below scroll thumb again, the ListView scrolls so that the second cell appears at the top. 
3) If you click above the scroll thumb, the ListView scrolls by pixel delta, as the second cell is also larger than the viewport. 

Now the ListView is in a different state than before the click (2).

These two behaviors are each useful for their own situation and should not be mixed.
1) Align the cell at the top or bottom if you have cells of the same size that are smaller than the viewport.
2) Scroll by pixel delta if you have cells of different sizes or if the cells are larger than the viewport.